install/kubernetes/cilium/templates/cilium-configmap.yaml

```diff
@@ -19,7 +19,7 @@
   {{- $defaultEnableCnpStatusUpdates = "false" -}}
   {{- $defaultBpfMapDynamicSizeRatio = 0.0025 -}}
   {{- $defaultBpfMasquerade = "true" -}}
-  {{- $defaultBpfClockProbe = "true" -}}
+  {{- $defaultBpfClockProbe = "false" -}}
```

Comment:
Not sure whether it's the right place to change it :-/ Maybe someone from the @cilium/helm could review?

Reply:
Yes, this is correct; in the daemon itself it defaults to false.

Reply:
I think Martynas wasn't asking Helm vs. agent, but this section of the Helm template vs. the section above.

Won't a change of time measurement for existing installations cause all existing connections to be interrupted? I would expect that's why we have this here to ensure the setting is preserved is the cluster was initially deployed with it, no?

Reply:
Yeah this logic here is very confusing. The way this `upgradeCompatibility`  section here works ([docs here](https://docs.cilium.io/en/stable/operations/upgrade/#step-2-use-helm-to-upgrade-your-cilium-deployment)) is that we have the old pre-v1.8 default in the top of the file, and then for each version where we change the default value, we add a new section.

So this section here is for default values changed in v1.8 (where we apparently turned bpf clock probe on by default). So I would argue it's not the right place to change. 

If we just want to turn it off by default again for new installations in v1.14+, we probably should add a v1.14 section where we do `$defaultBpfClockProbe = "false"`. If people then explicitly declare they want to use v1.12 default values, then bpf clock probe would still be enabled, as it was in v1.12.

If, on the other hand, we want to turn of bpf clock probe even for existing installations, then we should probably add it to the `values.yaml` and document this in the upgrade impact section of the docs. That way, users can set it back to `true` if they know what they are doing.

Reply:
I think the current understanding is that this feature had been effectively turned off everywhere until https://github.com/cilium/cilium/pull/25795. Quoting from the PR desc: 

> The Cilium agent has been throwing the 'Auto-disabling "enable-bpf-clock-probe" feature since kernel doesn't expose /proc/timer_list' warning for a while now.

(admittedly, I don't know how long this "for a while now" is)

Reply:
It seems it's already false in the values.yaml file, which I think was what Casey meant by his previous message:
https://github.com/cilium/cilium/blob/32118ccaa8677a3b836e60542a52a2af94693533/install/kubernetes/cilium/values.yaml#L428

Reply:
Hm. So I did a quick check (`helm template cilium ./install/kubernetes/cilium --namespace kube-system  | grep enable-bpf-clock-probe`) and we have a bug in our Helm charts. BPF probes *are currently turned on by default*, because the above value in `values.yaml` is not used and so we fall back on the v1.8 defaults.

The value in `values.yaml` is called `bpf.clockProbe`:
https://github.com/cilium/cilium/blob/32118ccaa8677a3b836e60542a52a2af94693533/install/kubernetes/cilium/values.yaml#L428

But the way it is set in the config map is using `bpfClockProbe` (no dot):
https://github.com/cilium/cilium/blob/ffc60dd087d55f0466772dba16710c11dbcf5c5d/install/kubernetes/cilium/templates/cilium-configmap.yaml#L297

I think we should rename the unused variable in `values.yaml` to `bpfClockProbe`, or make the configmap use both the defacto-used `bpfClockProbe` and the intended `bpf.clockProbe` value instead.

Reply:
> (admittedly, I don't know how long this "for a while now" is)

AFAIK, for a while is since that flag was added.

> I think the current understanding is that this feature had been effectively turned off everywhere until https://github.com/cilium/cilium/pull/25795.

Unfortunately, that fix got shipped into v1.13.4, which means some customers may be running with that feature enabled now.